tests/lib/rules/no-direct-mutation-state.js

```diff
@@ -70,6 +70,14 @@ ruleTester.run('no-direct-mutation-state', rule, {
       '}'
     ].join('\n')
   }, {
+    code: [
+      'class Hello extends React.Component {',
+      '  constructor() {',
+      '    this.state.foo = 1;',
```

Comment:
Oh shoot, I must have missed adding that line checking another thing… Sorry.

I had `this.state.foo++` right here as a valid case - we're in the constructor and while I think it's awkward, it's legal.